From 04f845a9d65826ac7e061d7b119f401e4de343c2 Mon Sep 17 00:00:00 2001
From: davidz-ampere <>
Date: Mon, 24 Mar 2025 22:52:47 -0400
Subject: [PATCH 1/1] Fix load in align_ldst_regoff

Data is not written back to the register and will cause data mismatch.
---
 arch/arm64/mm/fault.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/mm/fault.c b/arch/arm64/mm/fault.c
index e1d7dbaf711b..3de669a618a2 100644
--- a/arch/arm64/mm/fault.c
+++ b/arch/arm64/mm/fault.c
@@ -1032,6 +1032,7 @@ static int align_ldst_regoff(u32 insn, struct pt_regs *regs)
 			else
 				data = sign_extend64(data, datasize - 1);
 		}
+		pt_regs_write_reg(regs, t, data);
 	}
 
 	return 0;
-- 
2.27.0

